### PR TITLE
Search for pip-installed plugins when installing plugin imports

### PR DESCRIPTION
**Pull request by aaroncameron-wk**
#### Reason for change
Pip-installed plugins were not working as expected when pip-installed plugins imported other plugins (either pip-installed or built-in).

#### Description of change
Refactor plugin discovery by implementing `EntryPointRef` which handles the discovery, storage, and search of references to both built-in and installed plugins.

Update `freshenModules` to reload renamed plugins using the same process as missing plugins.

Implement plugin "aliases" to ease transitioning of built-in plugins to installed plugins.
This involved separating out the parsing of `__pluginInfo__` dictionaries from the recursive `moduleInfo` generation, which also means the "discovery" of plugins involves reading each of the discovered plugins to parse the "name" and "aliases" fields.

UI update to "Manage plugins" > "Select" menu. Removed nesting UI in favor of simple list of discovered plugins. Renamed some built-in plugins for consistency. Removed the `validate` plugin which was only used to import *all* validation plugins at once.

Screenshot 2023-11-15 at 12 13 28 PM

#### Steps to Test
For each of the following scenarios:
- Open the plugin Select menu (Help > Manage plug-ins > Select) and confirm options appear as expected.
- Run Arelle while invoking the associated plugin to confirm it's loaded correctly.

___
#### Default Setup
Keep all existing built-in plugins.
```
Edgar Renderer: not visible
Validate EFM: visible
```

___
#### Manually Installed Plugin
Install EdgarRenderer manually
```
Edgar Renderer: visible
Validate EFM: visible
```

___
#### Pip-installed Plugin
Pip-install [this branch](https://github.com/austinmatherne-wk/EdgarRenderer/tree/python-package) which provides `EdgarRenderer` and dependent plugins as entry points.
Delete `validate/EFM`, `validate/EFM-htm`, and `transforms` built-in plugin directories.
```
Edgar Renderer: visible
Validate EFM: visible
```

___
**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Review by austinmatherne-wk — DISMISSED** [review]

**Review by allenbross-wf — DISMISSED** [review]

File: arelle/PluginManager.py
Comment (aaroncameron-wk) [inline arelle/PluginManager.py:335]: I noticed `ixbrl-viewer` was showing as version 0.1, so I took the opportunity to update the version string with the pip package version if one was not found in the `__pluginInfo__` object.

Comment (austinmatherne-wk) [reply arelle/PluginManager.py:335]: I think this should be flipped. Use the entry point version if it exists, otherwise use the module info version

Comment (aaroncameron-wk) [reply arelle/PluginManager.py:340]: It makes sense to me to leave it up to the plugin to decide which to use. If the plugin maintainer intends for the package version to be used, it wouldn't make sense for them to be providing a hardcoded version. On the other hand, if they *don't* want the package version used, this allows them to override it.

**Review by austinmatherne-wk — DISMISSED** [review]

**Pull request merged**
